script.js

```diff
@@ -21,19 +26,107 @@ const addTodoHandler = () => {
 
 const deleteTodoHandler = (todoId) => {
     todos = todos.filter((todo) => todo.id !== todoId);
-
     renderTodos(todos);
 };
 
+const editTodoHandler = (
+    e,
+    inputElement,
+    paragraphElement,
+    cancelButton,
+    todo
+) => {
+    const $buttonElement = e.target;
+
+    if (!todo.isEditing) {
+        $buttonElement.innerText = "Update";
+        inputElement.value = todo.title;
+
+        displayPropertySetter(inputElement, "inline-block");
+        displayPropertySetter(cancelButton, "inline-block");
+        displayPropertySetter(paragraphElement, "none");
+
+        todo.isEditing = true;
+    } else {
+        if (!inputElement.value) {
+            alert("You can not update an empty title.");
+            return;
+        }
+
+        $buttonElement.innerText = "Edit";
+        paragraphElement.textContent = inputElement.value;
+
+        displayPropertySetter(inputElement, "none");
+        displayPropertySetter(cancelButton, "none");
+        displayPropertySetter(paragraphElement, "inline-block");
+
+        todo.title = inputElement.value;
+        todo.isEditing = false;
+    }
```

Comment:
1. Use early return.
2. Let's not use thread blocking calls like `alert`. Instead we can create error notification system later on as a bonus. For now, show the error in some place of the UI.
```suggestion
    if (!todo.isEditing) {
        $buttonElement.innerText = "Update";
        inputElement.value = todo.title;

        displayPropertySetter(inputElement, "inline-block");
        displayPropertySetter(cancelButton, "inline-block");
        displayPropertySetter(paragraphElement, "none");

        todo.isEditing = true;
        
        return
    }
    if (!inputElement.value) {
        showGlobalError("You can not update an empty title.");
        return;
     }

    $buttonElement.innerText = "Edit";
    paragraphElement.textContent = inputElement.value;

    displayPropertySetter(inputElement, "none");
    displayPropertySetter(cancelButton, "none");
    displayPropertySetter(paragraphElement, "inline-block");

     todo.title = inputElement.value;
     todo.isEditing = false;
```